accounts/ethstore/src/accounts_dir/disk.rs

```diff
@@ -117,16 +119,25 @@ pub struct DiskDirectory<T> where T: KeyFileManager {
 }
 
 /// Keys file manager for root keys directory
-pub struct DiskKeyFileManager;
+#[derive(Default)]
+pub struct DiskKeyFileManager {
+	password: Option<Password>,
+}
 
 impl RootDiskDirectory {
 	pub fn create<P>(path: P) -> Result<Self, Error> where P: AsRef<Path> {
 		fs::create_dir_all(&path)?;
 		Ok(Self::at(path))
 	}
 
+	/// allows to read keyfiles with given password (needed for keyfiles w/o address)
+	pub fn with_password(&self, password: Option<Password>) -> Self {
```

Comment:
I don't think it's a good idea to implement that here. Reasons:
1. The assumption is that all accounts inside the directory share the same password which is most likely incorrect.
2. It's used only for python keyfiles import, but will polute (already convoluted) codebase

Can we not use `DiskDirectory` for python keyfiles import and just allow importing particular account files?
Alternatively can't we treat python keyfiles as a vault (they share the same password actually) and use structs from this code part?

Reply:
ad1. I agree, it is not very generic, but as a user I just create special dir for my keyfiles (even if its single one) before import.

ad 2. it is used for all keyfiles w/o address. Not only pyethereum ones. As any new feature it introduces some noice. 

> Can we not use DiskDirectory for python keyfiles import and just allow importing particular account files?

I've tried to be consistent with current approach which is importing whole directories. Introduction of new `import-single` in CLI seems overkill to me.

>Alternatively can't we treat python keyfiles as a vault (they share the same password actually) and use structs from this code part?

I've analyzed vault's, but it does not fit here. You need to repack your keyfiles into valuts, and in orted to repack you need to import it first, right?

Reply:
ad1. if your keyfiles have different pwds you will see warnings, but import will succeed with at least this file with `pwd` you've entered. You can reexecute `import` with another `pwd` to import others.

Reply:
We have `import-wallet`, but it seems it only works for presale wallets currently. I'd be more inclined to handle it there than to hack around `DiskKeyFileManager` (since the former command allows importing directly to a vault, etc).

That said I'm fine with this if you address the `SafeAccount` semantics in a way I proposed in the other comment.